### ISSUE TITLE
Setup thread

### ISSUE THREAD

**Issue by karfau** [xmldom/is-dom-node#1]
- [x] lisense (suggestion MIT, to be aligned with xmldom)
- [x] ci (tests, type checks, linting, prettier, ...?)
- [x] release (just GitHub for now until npm is configurable?)

@cjbarth let's discuss stuff here to make it visible

**Comment by cjbarth** [comment]
I'm happy with MIT.
I'll probably just use mocha/chai for testing, istanbul/nyc for code coverage, eslint/prettier for code quality, and typescript/ts-node for TypeScript.

I'll probably also add some GitHub workflows for CodeQL and Dependabot along with checks against Node@ n-1, n, and n+1.

I'm happy doing GitHub for now, did you want to use the auto-changelog tool (my fork of GREN) for this along with release-it?

**Comment by karfau** [comment]
The one thing I would like to avoid is another manual release process. So if it can be automated it would be really valuable.
(I'm currently [using auto-changelog in xmldom in a very manual fashion](https://github.com/xmldom/xmldom/blob/master/docs/RELEASE.md#how-to-release) and [it will hopefully change to an automated process soon](https://github.com/xmldom/xmldom/issues/451), but there are quite some things to consider and decide.)

**Comment by cjbarth** [comment]
Using `release-it` I pretty much have an automated process. There is one more thing I can improve with changelog generation that would make it fully automatic. However, since it only takes literally 1 minute to do a release, I prefer to keep it manual in `node-saml` and `passport-saml`, and even `xml-crypto`; it keeps just one last set of eyes on the process to make sure everything goes right. Otherwise it is a "yes", "yes", "yes" kind of release answering script prompts.

I'll invest in the last little bit for the changelog generation, but I was wondering what automated release process you'd like to use because I'm not aware of any that I 100% trust to do the right thing (correct changelog, correct semver version bump, etc.).

Having said that, I'd like to see if we can make progress on this to alleviate some issue I and others are experiencing: https://github.com/goto100/xpath/issues/126. Once we get the first commit, the rest should go quickly since I've already pretty much written all the code and have the scaffolding from other projects.

**Comment by karfau** [comment]
Sounds great.

In general I just wanted to make you aware of the situation in xmldom, but it is nothing that should have any impact on this repo.

(More the other way around: looking forward to be inspired.)

**Comment by cjbarth** [comment]
Did you want to do an `npm init` and commit that so that everything is aligned with the way you'd like it? Then I'll start making PRs for the rest.

**Comment by karfau** [comment]
Sorry for the late response.
I hope you didn't wait for me to do that.
Since we are not able to publish to npm right now, I would expect you can run that command and if there is any issue I can file a PR.
My resources are quite low right now, so please go ahead for now.
I'm following the repo, so I will be aware what is happening, but not always able to respond within a day

**Comment by cjbarth** [comment]
Do we want to use the `main` or `master` branch for this?

**Comment by karfau** [comment]
I would prefer main

**Comment by cjbarth** [comment]
The first commit is up, and I'll start making PRs for the rest of the work. Did you want to review the PRs I make, or just let me review and land them myself?

**Comment by karfau** [comment]
I'm quite busy/overloaded right now, go ahead and do what you need to do.
If you want my feedback on something specific you can of course file a PR and ask for my review.

**Comment by cjbarth** [comment]
How is all the code looking now? Did you want me to try to do a tag and push to NPM? If you do it for the first one, even if there is no CHANGELOG, then we can do a semver-patch release that is an automated release with the CHANGELOG updated. I've updated my CHANGELOG building tools to hopefully do the right thing.

**Comment by karfau** [comment]
Sadly I still have not heard back from the current npm org owner.
We will have to stick to publishing on github for now until this is resolved. Or temporarily use a different scope to publish it if it is important...

**Comment by karfau** [comment]
@brodybits is there any chance for you to answer my questions/request any time soon, now that everything else is ready?

**Comment by karfau** [comment]
@cjbarth I approved your request for gitkraken.

I'm thinking of creating some community channel like slack or discord to be able to offer some more direct communication, without the need to file or comment on an issue.

Are you up for it?
Any suggestion/preference?

**Comment by cjbarth** [comment]
At `node-saml`, we use Slack, however, we only use it for maintainers. We really want people with questions to open an issue or Discussion so that the question and its answer become public knowledge. There are several that watch that project and contribute to issues and discussions but nothing else, and they are super helpful. They turn around and often point to other discussions and issues to help noobs get up to speed and answer questions they have which have already been answered.

So, long story short, I'm in favor of Slack, but only for maintainers.

**Comment by karfau** [comment]
https://join.slack.com/t/xmldomworkspace/shared_invite/zt-23aaz1vzq-~chW8an7AK999OoD6WXOOw

**Comment by karfau** [comment]
https://www.npmjs.com/package/@xmldom/is-dom-node
Time to celebrate and close this ticket.
Great work.
